File: PY/DAY2/alarm.py
def restore_code(content):
  content = content.split(',')
  codes = list(map(lambda x: int(x),content))
  limit = range(0,len(codes)-1,4)
  for index in limit: 
    num_1_pos = codes[index + 1]
    num_2_pos = codes[index + 2]
    output_pos = codes[index + 3]
    if codes[index] is 1:
      codes[output_pos] = codes[num_1_pos] + codes[num_2_pos]
      continue
    if codes[index] is 2:
      codes[output_pos] = codes[num_1_pos] * codes[num_2_pos]
  return codes[0]    

f = open('alarm_code.txt')
print(restore_code(f.read()))
f.close()

File: JS/DAY2/alarm.js
const fs = require('fs');

const updateCode = () => {
  for (let i = 0; i <= 99; i++) {
    for (let j = 0; j <= 99; j++) {
      const content = fs.readFileSync('DATA/alarmCode.txt', 'utf8');
      const codes = content.split(',').map(code => +code);
      codes[1] = i;
      codes[2] = j;
      const result = restoreCode(codes);
      if (result[0] === 19690720) {
        return result;
      }
    }
  }
};

const restoreCode = codes => {
  for (let index = 0; index < codes.length; index += 4) {
    const num1Pos = codes[index + 1];
    const num2Pos = codes[index + 2];
    const outputPos = codes[index + 3];
    if (codes[index] === 1) {
      codes[outputPos] = codes[num1Pos] + codes[num2Pos];
      continue;
    }
    if (codes[index] === 2) {
      codes[outputPos] = codes[num1Pos] * codes[num2Pos];
    }
  }
  return codes;
};

const main = () => {
  const result = updateCode();
  console.log(result[1], result[2]);
};

main();
